model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1146,33 +1146,41 @@ message StandardArtifacts {
   enum Types {
     // A URN for locally-accessible artifact files.
     // payload: ArtifactFilePayload
-    FILE     = 0 [(beam_urn) = "beam:artifact:type:file:v1"];
+    FILE      = 0 [(beam_urn) = "beam:artifact:type:file:v1"];
 
     // A URN for artifacts described by URLs.
     // payload: ArtifactUrlPayload
-    URL      = 1 [(beam_urn) = "beam:artifact:type:url:v1"];
+    URL       = 1 [(beam_urn) = "beam:artifact:type:url:v1"];
 
     // A URN for artifacts embedded in ArtifactInformation proto.
     // payload: EmbeddedFilePayload.
-    EMBEDDED = 2 [(beam_urn) = "beam:artifact:type:embedded:v1"];
+    EMBEDDED  = 2 [(beam_urn) = "beam:artifact:type:embedded:v1"];
 
     // A URN for Python artifacts hosted on PYPI.
     // payload: PypiPayload
-    PYPI     = 3 [(beam_urn) = "beam:artifact:type:pypi:v1"];
+    PYPI      = 3 [(beam_urn) = "beam:artifact:type:pypi:v1"];
 
     // A URN for Java artifacts hosted on a Maven repository.
     // payload: MavenPayload
-    MAVEN    = 4 [(beam_urn) = "beam:artifact:type:maven:v1"];
+    MAVEN     = 4 [(beam_urn) = "beam:artifact:type:maven:v1"];
+
+    // A URN for locally-accessible artifact directory.
+    // payload: ArtifactDirectoryPayload
+    DIRECTORY = 5 [(beam_urn) = "beam:artifact:type:directory:v1"];
   }
   enum Roles {
     // A URN for staging-to role.
     // payload: ArtifactStagingToRolePayload
     STAGING_TO  = 0 [(beam_urn) = "beam:artifact:role:staging_to:v1"];
+
+    // A URN for unzip-to role.
+    // payload: ArtifactUnzipToRolePayload
+    UNZIP_TO    = 1 [(beam_urn) = "beam:artifact:role:unzip_to:v1"];
   }
 }
 
 message ArtifactFilePayload {
-  // a string for an artifact path e.g. "/tmp/foo.jar"
+  // a string for an artifact file path e.g. "/tmp/foo.jar"
   string path = 1;
 
   // The hex-encoded sha256 checksum of the artifact.
```

Comment:
It doesn't seem like we are populating sha256 here.

Reply:
adding deferred artifacts and populating sha256.